### PR TITLE
metadata post ga changes in instance

### PR DESCRIPTION
**Pull request by deepaksibm**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: website/docs/r/is_instance.html.markdown [hunk 1]
```diff
@@ -340,6 +340,15 @@ Review the argument references that you can specify for your resource.
   `image` conflicts with `boot_volume.0.snapshot`, not required when creating instance using `instance_template`
 - `keys` - (Required, List) A comma-separated list of SSH keys that you want to add to your instance.
 - `metadata_service_enabled` - (Optional, Boolean) Indicates whether the metadata service endpoint is available to the virtual server instance. Default value : **false**
+
+  ~> **NOTE**
+  `metadata_service_enabled` is deprecated and will be removed in the future. Use `metadata_service` instead
+- `metadata_service` - (Optional, String) The metadata service configuration.
```
Comment (astha-jain) [inline website/docs/r/is_instance.html.markdown:346]: (Optional, String) - not a string

**Comment by deepaksibm** [comment]
In prod  -  vpcdemo
```

$ tfpt TestAccIBMISInstance_metadata_service
=== RUN   TestAccIBMISInstance_metadata_service
no message sent


no message sent
--- PASS: TestAccIBMISInstance_metadata_service (364.52s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     366.077s
```

File: ibm/service/vpc/resource_ibm_is_instance_template.go [hunk 2]
```diff
@@ -107,13 +107,51 @@ func ResourceIBMISInstanceTemplate() *schema.Resource {
 			},
 
 			isInstanceTemplateMetadataServiceEnabled: {
-				Type:        schema.TypeBool,
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/resource_ibm_is_instance_template.go:110]: Can we add acceptance test for instance template also?

**Review by SunithaGudisagarIBM — COMMENTED** [review]
Please find my comment Deepak

Thank you
Sunitha

**Comment by SunithaGudisagarIBM** [comment]
Please do make fmt and push the code
Screenshot 2023-02-01 at 3 37 03 PM

File: ibm/acctest/acctest.go [hunk 3]
```diff
@@ -516,7 +516,7 @@ func init() {
 	IsImage = os.Getenv("IS_IMAGE")
 	if IsImage == "" {
 		//IsImage = "fc538f61-7dd6-4408-978c-c6b85b69fe76" // for classic infrastructure
-		IsImage = "r006-13938c0a-89e4-4370-b59b-55cd1402562d" // for next gen infrastructure
+		IsImage = "r134-5800d6ef-ad8b-464e-b242-e69e43b93aea" // for next gen infrastructure
```
Comment (SunithaGudisagarIBM) [inline ibm/acctest/acctest.go:519]: Revert changes once testing is done

**Comment by SunithaGudisagarIBM** [comment]
```
resource "ibm_is_instance" "testacc_instance" {
  name    = "sunitha-image"
  image   = "r006-b4d8d8d4-7768-4a1b-9434-9483ca821a0d"
  profile = "cx2-2x4"
  primary_network_interface {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
  }
  vpc  = "r006-0d37163a-701d-4ad6-9ece-a3e34cf28935"
  zone = "us-south-1"
  keys = ["r006-d3f8e88f-1adb-4c4b-973c-47dc68097343"]
  network_interfaces {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
    name   = "eth1"
  }
  metadata_service {
    response_hop_limit = 2
  }
}
```

Screenshot 2023-02-01 at 5 58 10 PM
Screenshot 2023-02-01 at 5 58 24 PM

**Comment by SunithaGudisagarIBM** [comment]
```
resource "ibm_is_instance_template" "instancetemplate1" {
  name    = "sunitha-instance-temp"
  image   = "r006-b4d8d8d4-7768-4a1b-9434-9483ca821a0d"
  profile = "cx2-2x4"

  primary_network_interface {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
  }

  vpc  = "r006-0d37163a-701d-4ad6-9ece-a3e34cf28935"
  zone = "us-south-1"
  keys = ["r006-d3f8e88f-1adb-4c4b-973c-47dc68097343"]
}
```

File: ibm/service/vpc/data_source_ibm_is_instance_template.go [hunk 4]
```diff
@@ -120,6 +120,32 @@ func DataSourceIBMISInstanceTemplate() *schema.Resource {
 				Computed:    true,
 				Description: "Indicates whether the metadata service endpoint is available to the virtual server instance",
 			},
+			isInstanceMetadataService: {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_instance_template.go:123]: Please add the schema for metadata service here

File: ibm/service/vpc/data_source_ibm_is_instance_templates.go [hunk 5]
```diff
@@ -151,6 +151,7 @@ func DataSourceIBMISInstanceTemplates() *schema.Resource {
 							Computed:    true,
 							Description: "The unique identifier or CRN of the default IAM trusted profile to use for this virtual server instance.",
 						},
+
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_instance_templates.go:154]: Please add the schema for metadata service here

Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/data_source_ibm_is_instance_templates.go:154]: Templates is not returning anything please check

 data.ibm_is_instance_templates.instance_templates_data:
data "ibm_is_instance_templates" "instance_templates_data" {
    id = "2023-02-01 12:43:06.68559 +0000 UTC"
}

**Review by SunithaGudisagarIBM — CHANGES_REQUESTED** [review]
Please my comments

Thanks and Regards
Sunitha

File: ibm/service/vpc/data_source_ibm_is_instance_templates.go (hunk 5 above)
Comment (SunithaGudisagarIBM) [reply ibm/service/vpc/data_source_ibm_is_instance_templates.go:154]: There are multiple places you need to add the logic, ex : line number 770, Please check in all the data sources,
As of now the values are not getting populated in any of the data sources

				if instance.MetadataService != nil {
					d.Set(isInstanceTemplateMetadataServiceEnabled, instance.MetadataService.Enabled)
				}

File: ibm/service/vpc/data_source_ibm_is_instance.go [hunk 6]
```diff
@@ -59,7 +59,32 @@ func DataSourceIBMISInstance() *schema.Resource {
 				Computed:    true,
 				Description: "Indicates whether the metadata service endpoint is available to the virtual server instance",
 			},
+			isInstanceMetadataService: {
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_instance.go:62]: As of now the values are not getting populated in any of the data sources, Please review the changes again

File: ibm/service/vpc/data_source_ibm_is_instances.go [hunk 7]
```diff
@@ -121,6 +121,32 @@ func DataSourceIBMISInstances() *schema.Resource {
 							Computed:    true,
 							Description: "Indicates whether the metadata service endpoint is available to the virtual server instance",
 						},
+						isInstanceMetadataService: {
+							Type:        schema.TypeList,
```
Comment (SunithaGudisagarIBM) [inline ibm/service/vpc/data_source_ibm_is_instances.go:125]: As of now the values are not getting populated in any of the data sources, Please review the changes again

**Review by SunithaGudisagarIBM — CHANGES_REQUESTED** [review]

**Comment by deepaksibm** [comment]
```
$ tfpt TestAccIBMISInstanceTemplate_metadata_service
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./ibm/service/vpc -v -run=TestAccIBMISInstanceTemplate_metadata_service -timeout 700m
=== RUN   TestAccIBMISInstanceTemplate_metadata_service
--- PASS: TestAccIBMISInstanceTemplate_metadata_service (825.51s)
PASS
ok      github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/vpc     827.074s
```

**Comment by deepaksibm** [comment]
address all the comments

**Comment by SunithaGudisagarIBM** [comment]
After Review 
```
resource "ibm_is_instance" "testacc_instance" {
  name    = "sunitha-image"
  image   = "r006-b4d8d8d4-7768-4a1b-9434-9483ca821a0d"
  profile = "cx2-2x4"
  primary_network_interface {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
  }
  vpc  = "r006-0d37163a-701d-4ad6-9ece-a3e34cf28935"
  zone = "us-south-1"
  keys = ["r006-720a844d-4083-4353-a4d5-fe055737deef"]
  network_interfaces {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
    name   = "eth1"
  }
}
data "ibm_is_instances" "ds_instances" {
}

data "ibm_is_instance" "ds_instance" {
  name = "sunitha-image"
}
```
Resource -
Screenshot 2023-02-06 at 3 46 20 PM

Datasource -
Screenshot 2023-02-06 at 3 46 53 PM
Screenshot 2023-02-06 at 3 47 26 PM

**Comment by SunithaGudisagarIBM** [comment]
Template test after code review changes :

```
resource "ibm_is_instance_template" "instancetemplate1" {
  name    = "sunitha-instance-temp"
  image   = "r006-b4d8d8d4-7768-4a1b-9434-9483ca821a0d"
  profile = "cx2-2x4"

  primary_network_interface {
    subnet = "0717-cd4f74f1-bfae-423a-886b-9163256ec4a9"
  }

  vpc  = "r006-0d37163a-701d-4ad6-9ece-a3e34cf28935"
  zone = "us-south-1"
  keys = ["r006-720a844d-4083-4353-a4d5-fe055737deef"]
  metadata_service {
    enabled            = true
    protocol           = "http"
    response_hop_limit = 2
  }
}

data "ibm_is_instance_template" "instance_template_data" {
  name = "sunitha-instance-temp"
}

data "ibm_is_instance_templates" "instance_templates_data" {
}
```

resource output:
Screenshot 2023-02-06 at 3 49 44 PM

datasource output:
Screenshot 2023-02-06 at 3 50 56 PM

Screenshot 2023-02-06 at 3 51 11 PM

**Review by SunithaGudisagarIBM — APPROVED** [review]
Hi Deepak, 

Code changes seems to be fine for me.

Thank you
Sunitha

**Comment by deepaksibm** [comment]
going upstream

**Pull request closed without merging**